pallets/proofs-dealer/src/lib.rs

```diff
@@ -429,37 +479,20 @@ pub mod pallet {
             // If the proof is valid, the execution of this extrinsic should be refunded.
             Ok(Pays::No.into())
         }
+    }
 
-        /// Extrinsic to register a new round of challenges.
-        ///
-        /// This function is called by the block producer to register a new round of challenges.
-        /// Random challenges are automatically generated based on some external source of
-        /// randomness, and are added to `BlockToChallenges`, for this block's number.
-        ///
-        /// It also takes care of including the challenges from the `ChallengesQueue` and
-        /// `PriorityChallengesQueue`. This custom challenges are only included in "checkpoint"
-        /// blocks
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        /// This hook is used to generate new challenges.
         ///
-        /// Additionally, it takes care of checking if there are Providers that have
-        /// failed to submit a proof, and should have submitted one by this block. It does so
-        /// by checking the `BlockToChallengedProviders` StorageMap. If a Provider is found
-        /// to have failed to submit a proof, it is subject to slashing.
-        ///
-        /// Finally, it cleans up:
-        /// - The `BlockToChallenges` StorageMap, removing entries older than `ChallengeHistoryLength`.
-        /// - The `BlockToChallengedProviders` StorageMap, removing entries for the current block number.
-        #[pallet::call_index(2)]
-        #[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-        pub fn new_challenges_round(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            ensure_none(origin)?;
-
-            // TODO: Handle result of verification.
-            Self::do_new_challenges_round()?;
-
-            // TODO: Emit events.
-
-            // Return a successful DispatchResultWithPostInfo
-            Ok(().into())
+        /// It will be called at the beginning of every block, if the block is not being part of a
+        /// [Multi-Block-Migration](https://github.com/paritytech/polkadot-sdk/pull/1781) (MBM).
+        /// For more information on the lifecycle of the block and its hooks, see the [Substrate
+        /// documentation](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.Hooks.html#method.on_poll).
+        fn on_poll(n: BlockNumberFor<T>, weight: &mut frame_support::weights::WeightMeter) {
```

Comment:
My understanding is that it would not be executed only during a Multi Block Migration (see [here](https://github.com/paritytech/polkadot-sdk/pull/1781)). That's why I keep the `CurrentTicker` instead of just looking at block numbers. The `CurrentTicker` is just like the block number, except that it doesn't get incremented when `on_poll` is not executed